test/e2e/framework/network/utils.go

```diff
@@ -755,6 +770,11 @@ func (config *NetworkingTestConfig) setup(selector map[string]string) {
 		}
 	}
 	config.ClusterIP = config.NodePortService.Spec.ClusterIP
+	if config.DualStackEnabled {
+		config.SecondaryClusterIP = config.NodePortService.Spec.ClusterIPs[1]
+	}
+	// TODO: add SecondaryNodeIP configuration once it is implemented
+	// xref: https://github.com/kubernetes/enhancements/pull/1665
```

Comment:
Should we check if the secondary internal IP exists and use that for tests? For cloud providers, the Internal IPs for each family does get populated

```
Addresses:
  Hostname:    k8s-linuxpool1-15729367-vmss000000
  InternalIP:  10.240.0.4
  InternalIP:  2001:1234:5678:9abc::5
```

Similar to this check - https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/network/utils.go#L758-L762, we can validate if second address exists for NodeInternalIP and use it if it belongs to a different family?

Reply:
we can do it for all, I have to modify the test to support it. It was an oversight on my side, thanks for catching it